homeassistant/components/fan/__init__.py

```diff
@@ -238,6 +240,9 @@ def set_direction(self: ToggleEntity, direction: str) -> None:
 
     def turn_on(self: ToggleEntity, speed: str=None, **kwargs) -> None:
         """Turn on the fan."""
+        if speed is SPEED_OFF:
+            self.turn_off()
```

Comment:
Isn't it weird to have this in the base component? How would a platform call this, without risk of raising `NotImplementedError`? It would have to be done within an IF statement, and then what's the point? The platform could just imlement all the logic directly